tools/state-viewer/src/state_dump.rs

```diff
@@ -65,6 +65,10 @@ pub fn state_dump(
         })
         .collect();
     genesis_config.validators.sort_by_key(|account_info| account_info.account_id.clone());
+    if let Some(_) = select_account_ids {
+        // Keep at most one validator outside of selected accounts if account selection is enabled.
+        genesis_config.validators = vec![genesis_config.validators[0].clone()];
```

Comment:
Can you add another argument for the number of validators you want to keep? This is hidden inside the logic and has nothing to do with select_account

Reply:
After thinking for a while, I think it might be better to move this change to a separate PR.

Reply:
Going to remove this one for now.